File: src/complex.cpp
#include "complex.h"

void In(complex &num, std::ifstream &in) {
    in >> num.a >> num.b;
}

void InRnd(complex &num) {
    num.a = rand() % 1000;
    num.b = rand() % 1000 + 1; 
}

void Out(complex &t, std::ofstream &out) {
    out << t.a << (t.b >= 0 ? " + " : " - ") << abs(t.b) << "i ";
}

double RationalRepr(complex &t) {
    return std::sqrt(t.a * t.a + t.b * t.b);
}

File: hw3/src/complex.py
import math


class complex:
    """
        Complex Number
        Num = a + b*i
    """
    def __init__(self, a, b):
        self.a = float(a)
        self.b = float(b)

    def rational_repr(self):
        return math.sqrt(self.a ** 2 + self.b ** 2)

    def __str__(self):
        if self.b > 0:
            return f"{self.a} + {self.b}i"
        else:
            return f"{self.a} - {abs(self.b)}i"
